administrator/components/com_templates/Model/TemplateModel.php

```diff
@@ -148,6 +154,152 @@ public function getDirectoryTree($dir)
 		return $result;
 	}
 
+	/**
+	 * Method to get the core file of override file
+	 *
+	 * @param   string  $file  Override file
+	 * @param   mixed  $client Client Id
+	 *
+	 * @return  string  $corefile The full path and file name for the target file, or boolean false if the file is not found in any of the paths.
+	 *
+	 * @since   4.0
+	 */
+	public function loadCoreFile($file, $client_id)
+	{
+		$app           = Factory::getApplication();
+		$filePath      = base64_decode($file);
+		$explodeArray  = explode(DIRECTORY_SEPARATOR, $filePath);
+		$size          = count($explodeArray);
+
+		// Only allow html/ folder
+		if($explodeArray['1'] !== 'html')
+		{
+			$app->enqueueMessage(Text::_('COM_TEMPLATES_ERROR_ONLY_HTML_FOLDER_ALLOWED'), 'error');
+
+			return false;
+		}
+
+		$fileName = end($explodeArray);
```

Comment:
You can use basename() here.

Reply:
Is basename safe ?

Reply:
It is a core function, so hopefully it is.